test/fake_http_server.h

```diff
@@ -94,8 +98,10 @@ class FakeServer {
     // take ownership.
     class Stream {
      public:
+      Stream() : connection_counter_(0) {}
       void AddQueue(std::queue<std::string>* queue);
-      bool WaitForOneWatcher(time::seconds timeout);
+      void RemoveQueue(std::queue<std::string>* queue);
+      bool WaitForConnectionCounter(int min_connections, time::seconds timeout);
```

Comment:
Shouldn't this also be `WaitForMinTotalConnections`?

Reply:
Done.